### PR TITLE
ConfigProvider support `csp` prop

### PR DESCRIPTION
**Pull request by zombieJ**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

#12438
  
### API Realization (Optional if not new feature)

```jsx
<ConfigProvider csp={{ nonce: 'YourNonceCode' }}>
  <Button>My Button</Button>
</ConfigProvider>
```

### What's the effect? (Optional if not new feature)

No affect

### Changelog description (Optional if not new feature)

> 1. ConfigProvider support Content Security Policy (CSP) config
> 2. ConfigProvider 支持 Content Security Policy (CSP) 配置

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by afc163 — APPROVED** [review]

File: components/config-provider/index.zh-CN.md [hunk 1]
```diff
@@ -24,6 +24,16 @@ return (
...
+
+```jsx
+<ConfigProvider csp={{ nonce: 'YourNonceCode' }}>
+  <Button>My Button</Button>
+</ConfigProvider>
+```
+
 ## API
 
 | 参数 | 说明 | 类型 | 默认值 |
```
Comment (afc163) [inline components/config-provider/index.zh-CN.md]: API 里没加 csp 的说明。

**Comment by liuhuiahwy** [comment]
https://github.com/ant-design/ant-design/pull/14222/files#diff-0f7d36c481fbb3385acdfa98a8c4cdc2R60

This line seems should change to 
'styleForPesudo.setAttribute('nonce', this.csp.nonce);'

@afc163

**Comment by zombieJ** [comment]
@liuhuiahwy , it's ok to set `nonce` directly: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/nonce#Syntax

**Comment by liuhuiahwy** [comment]
@zombieJ To get the value, it will be ok, but if setting the value it seems not working
image
And for the website that has allows the CSP, it will not work.

**Comment by zombieJ** [comment]
It does work. Check `nonce` prop instead of check hole element:
屏幕快照 2019-06-24 上午11 56 12

**Comment by liuhuiahwy** [comment]
@zombieJ Yeah, you are right. Check again. thanks.

**Comment by pkyeck** [comment]
Thanks for this PR!

But I don't really get how I'm supposed to use this? Do I have to wrap every `<Button>` with a `<ConfigProvider>`. A real example would help a lot, I think.

**Comment by zombieJ** [comment]
@pkyeck , ConfigProvider should always be the top of the web app. It use context to pass date all sub component under this will get `scp` content:
```jsx
<ConfigProvider>
  ...
</ConfigProvider>
```

**Pull request merged**